Sources/Benchmark/BenchmarkReporter.swift

```diff
@@ -12,22 +12,50 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import Dispatch
 import Foundation
 
+#if os(Linux)
+  import Glibc
+#else
+  import Darwin
+#endif
+
 protocol BenchmarkReporter {
-    func report(running name: String, suite: String)
-    func report(results: [BenchmarkResult])
+    mutating func report(running name: String, suite: String)
+    mutating func report(finishedRunning name: String, suite: String)
+    mutating func report(results: [BenchmarkResult])
 }
 
 struct PlainTextReporter: BenchmarkReporter {
-    func report(running name: String, suite: String) {
+    var benchmarkRunningState: (name: String, suite: String, startNanos: UInt64)? = nil
+
+    mutating func report(running name: String, suite: String) {
+        assert(benchmarkRunningState == nil, "benchmark running state: \(benchmarkRunningState!)")
         let prefix: String
         if suite != "" {
             prefix = "\(suite): "
         } else {
             prefix = ""
         }
-        print("running \(prefix)\(name)")
+        print("running \(prefix)\(name)...", terminator: "")
+        fflush(stdout)  // Flush stdout to actually see the message...
+        benchmarkRunningState = (name, suite, DispatchTime.now().uptimeNanoseconds)
+    }
+
+    mutating func report(finishedRunning name: String, suite: String) {
+        let end = DispatchTime.now().uptimeNanoseconds
```

Comment:
So far all of the measurements have been done in the main benchmark loop. The reporter only gets the recorded results, doesn't perform any measurements itself.

I think it would be good to preserve this flow of control, although I don't feel very strongly about it.

Reply:
Fixed in follow-up commit. SGTM for now.